pkg/controller/operators/openshift/helpers.go

```diff
@@ -105,8 +105,12 @@ func (s skew) String() string {
 	if s.err != nil {
 		return fmt.Sprintf("%s/%s has invalid %s properties: %s", s.namespace, s.name, MaxOpenShiftVersionProperty, s.err)
 	}
-
-	return fmt.Sprintf("%s/%s is incompatible with OpenShift minor versions greater than %s", s.namespace, s.name, s.maxOpenShiftVersion)
+	// is it safe to ignore the error here, with the assumption
+	// that we build a skew object only after verifying that the
+	// version string is parseable safely.
+	maxOCPVersion, _ := semver.ParseTolerant(s.maxOpenShiftVersion)
+	nextY := nextY(maxOCPVersion).String()
+	return fmt.Sprintf("ClusterServiceVersions blocking upgrade to %s or higher. The maximum supported OCP version for %s/%s is %s", nextY, s.namespace, s.name, s.maxOpenShiftVersion)
```

Comment:
For ref, [this](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/openshift/helpers.go#L144-L168) is where the skew objects are created, where [s.maxOpenshiftVersion](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/openshift/helpers.go#L165) is set AFTER the value of the field from the CSV is parsed [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/openshift/helpers.go#L154), and `maxOpenshiftVersion()` [checks that the value is parseable and returns the error if it's not](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/openshift/helpers.go#L281-L284)